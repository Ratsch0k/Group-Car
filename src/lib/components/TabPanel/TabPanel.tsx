import React, {PropsWithChildren} from 'react';

/**
 * Props for TabPanel
 */
export interface TabPanelProps {
  /**
   * Whether or not this panel is currently shown.
   */
  visible: boolean;
  id?: string;
  'aria-labelledby'?: string;
  className?: string;
}

/**
 * Tab panel for `Tabs`.
 * @param props Props
 */
export const TabPanel =
// eslint-disable-next-line react/display-name
React.forwardRef<
HTMLDivElement,
PropsWithChildren<TabPanelProps>
>((props, ref) => {
  return (
    <div
      className={props.className}
      role='tabpanel'
      hidden={!props.visible}
      id={props.id}
      aria-labelledby={props['aria-labelledby']}
      ref={ref}
    >
      {props.visible && props.children}
    </div>
  );
});

export default TabPanel;
