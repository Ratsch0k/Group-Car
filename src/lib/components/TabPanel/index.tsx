export * from './TabPanel';
export {default} from './TabPanel';
