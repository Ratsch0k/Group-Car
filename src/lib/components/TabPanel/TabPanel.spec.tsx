import React from 'react';
import { render } from "@testing-library/react";
import TabPanel from "./TabPanel";

it('renders children if visible is true', () => {
  const screen = render(
    <TabPanel visible id='test tabpanel' aria-labelledby='test tab'>
      <span>THIS IS JUST A TEST</span>
    </TabPanel>
  );

  expect(screen.baseElement).toMatchSnapshot();
});

it('doesn\'t render children if visible is false', () => {
  const screen = render(
    <TabPanel visible={false} id='test tabpanel' aria-labelledby='test tab'>
      <span>TEST</span>
    </TabPanel>
  );

  expect(screen.baseElement).toMatchSnapshot();
  expect(screen.queryByText('TEST')).toBeFalsy();
});