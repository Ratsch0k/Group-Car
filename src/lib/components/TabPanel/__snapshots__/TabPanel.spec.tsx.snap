// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't render children if visible is false 1`] = `
<body>
  <div>
    <div
      aria-labelledby="test tab"
      hidden=""
      id="test tabpanel"
      role="tabpanel"
    />
  </div>
</body>
`;

exports[`renders children if visible is true 1`] = `
<body>
  <div>
    <div
      aria-labelledby="test tab"
      id="test tabpanel"
      role="tabpanel"
    >
      <span>
        THIS IS JUST A TEST
      </span>
    </div>
  </div>
</body>
`;
